api/src/test/java/org/apache/iceberg/transforms/TestStartsWith.java

```diff
@@ -67,8 +67,8 @@ public void testTruncateString() {
     UnboundPredicate<String> projected = trunc.project(COLUMN, boundExpr);
     Evaluator evaluator = new Evaluator(SCHEMA.asStruct(), projected);
 
-    Assert.assertTrue("startsWith(abcde, truncate(abcde,2))  => true",
-        evaluator.eval(TestHelpers.Row.of("abcde")));
+    Assert.assertTrue("startsWith(abcde, truncate(abcdg,2))  => true",
+        evaluator.eval(TestHelpers.Row.of("abcdg")));
```

Comment:
I updated this test from abcde, as it seems much more revealing of a test that we're going to read a partition that actually does not start with the given value, because of the truncation width used in the partition spec.

I'd be happy to move this to anther PR given that it is relatively unrelated, or remove it if anybody feels that strongly about it,. It does seem to me to be much more informative of a test case than`abcde startsWith abcde`, as that would return true on an identity partition spec as well.